VisualPinball.Unity/VisualPinball.Unity/VPT/CollisionSwitch/CollisionSwitchApi.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using Logger = NLog.Logger;
+using NLog;
+using UnityEngine;
+using Unity.Entities;
+
+namespace VisualPinball.Unity
+{
+	public class CollisionSwitchApi : IApi, IApiSwitch, IApiSwitchDevice
+	{
+		private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+		private readonly CollisionSwitchComponent _collisionSwitchComponent;
+		private IColliderComponent _colliderComponent;
+		private Player _player;
+		private BallManager _ballManager;
+	
+		private protected readonly SwitchHandler SwitchHandler;
+
+		public event EventHandler Init;
+		public event EventHandler<SwitchEventArgs> Switch;
+
+		public bool IsSwitchEnabled => SwitchHandler.IsEnabled;
+		IApiSwitchStatus IApiSwitch.AddSwitchDest(SwitchConfig switchConfig) => SwitchHandler.AddSwitchDest(switchConfig.WithPulse(true));
+		void IApiSwitch.AddWireDest(WireDestConfig wireConfig) => SwitchHandler.AddWireDest(wireConfig.WithPulse(true));
+		void IApiSwitch.RemoveWireDest(string destId) => SwitchHandler.RemoveWireDest(destId);
+		IApiSwitch IApiSwitchDevice.Switch(string deviceItem) => this;
+		public void OnSwitch(bool closed) => SwitchHandler.OnSwitch(closed);
+		public void DestroyBall(Entity ballEntity) => _ballManager.DestroyEntity(ballEntity);
+		
+		internal CollisionSwitchApi(GameObject go, Player player)
+		{
+			_collisionSwitchComponent = go.GetComponentInChildren<CollisionSwitchComponent>();
+			_player = player;
+
+			SwitchHandler = new SwitchHandler(go.name, player);
+		}
+
+		void IApi.OnInit(BallManager ballManager)
+		{
+			_ballManager = ballManager;
+
+			_colliderComponent = _collisionSwitchComponent.GetComponentInParent<IColliderComponent>();
+
+			if (_colliderComponent is SurfaceColliderComponent) {
+				_player.TableApi.Surface(_collisionSwitchComponent.name).Hit += OnHit;
+			}
+			else if (_colliderComponent is RubberColliderComponent) {
+				_player.TableApi.Rubber(_collisionSwitchComponent.name).Hit += OnHit;
+			}
+			else if (_colliderComponent is PrimitiveColliderComponent) {
+				_player.TableApi.Primitive(_collisionSwitchComponent.name).Hit += OnHit;
+			}
+			else {
+				Logger.Error($"{_collisionSwitchComponent.name} not connected to a surface, rubber, or primitive");
+			}

```

Comment:
I'd prefer if this was interfaced so we can easily add other components in the future. For the API classes there is `IApiHittable` which contains the `OnHit` events. You'll probably have to keep a hittable component dictionary in `TableApi`, but I'd prefer to do it there, because it's a more central place we're less likely to miss in the future when adding other components.

Reply:
Okay, I will make the changes tonight.

Should we rename again to HitSwitch?

Reply:
I'd prefer Collision, Hit is still from the VPX code base, and I'd rather move to Collision, which seem to be the common term for that.

Reply:
Cool. Thx!